bash_archiving.sh

```diff
@@ -1,117 +1,124 @@
 #!/bin/bash
-USERS=$(cat /etc/passwd | sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' | tr '\n' ' ') ;
-countUSERS=$(wc -l /etc/passwd | grep -o '[0-9]*');
+USERS=$(sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' /etc/passwd | tr '\n' ' ')
```

Comment:
Programs like `sed` are already capable of reading a file, so it's entirely redundant to pipe into it the contents of a file. This sort of thing comes up a lot, so I'll only mention it here.